wslpy/system.py

```diff
@@ -78,4 +78,67 @@ def shellEnvVar(input):
 
 
 def registry(input, key):
-    raise NotImplementedError
\ No newline at end of file
+
+    #DUMMY CODE BEGINS (i am only using this to illustrate something)
+    cmd = u"reg.exe query \""+input+"\" /v \""+key+u"\" 2>&1"
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+    routput, err = p.communicate()
+    #DUMMY CODE ENDS 
+
+    # Given a valid registry path, retrieves the value of an entry in the registry.
+    # Eg: registry("HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\Environment","OS") returns "WINDOWS_NT"
+    #
+    # MAY return string type TODO: Doees this make sense?
+    #
+    # A valid registry path typically includes (this this and this)
+    #
+    # Parameters
+    # ----------
+    # input : str
+    #     string of a shell environment variable key.
+    #   
+    #     Currently expected values: 
+    #     "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\Environment" (for system)
+    #     "HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\User Shell Folders" (for shell)
+    #     (Although any valid entries should work too) TODO: Check if required
+    #
+    # key : str 
+    #     the name of the registry's key in string
+    #
+    # Returns
+    # -------
+    # The corresponding value TODO: Possibly type
+    #
+    # Raises
+    # ------
+    # Returns the error from Reg.exe (Possibly? )
+
+    try:
+        query = __regInfoFetch__(input, key)
+        
+        if query != "value.": #Error after regex #TODO: Check with Patrick on how he wants to handle this
+            return query #Expected one
+        else:
+            raise KeyError("The following error propogated from Registry:" + routput.decode("utf-8").rstrip()) #TODO: 
+    
+    except KeyError as err:
+        print(err)
+ 
+#### Test Cases
```

Comment:
These are for my own reference, I'll delete it when I have a final version ready to merge

Reply:
test cases should be placed under test in the future, currently, there is no test but I will hook up test suite to help you make it work -- once you are done